app/src/main/java/com/futurice/freesound/feature/search/SearchViewModel.java

```diff
@@ -100,16 +100,16 @@ void openSoundDetails(@NonNull final Sound sound) {
 
     @Override
     public void bind(@NonNull final CompositeDisposable disposables) {
-        disposables.add(searchTermSubject.observeOn(Schedulers.computation())
-                                         .map(String::trim)
-                                         .debounce(SEARCH_DEBOUNCE_TIME_SECONDS, TimeUnit.SECONDS)
-                                         .switchMap(this::searchOrClear)
-                                         .subscribe(nothing1(),
+        disposables.add(searchTermOnceAndStream.observeOn(Schedulers.computation())
+                                               .map(String::trim)
+                                               .debounce(SEARCH_DEBOUNCE_TIME_SECONDS, TimeUnit.SECONDS)
+                                               .switchMap(this::searchOrClearOnce)
+                                               .subscribe(nothing1(),
                                                     e -> e(e, "Error when setting search term")));
     }
 
     @NonNull
-    private Observable<Unit> searchOrClear(@NonNull final String searchQuery) {
+    private Observable<Unit> searchOrClearOnce(@NonNull final String searchQuery) {
         return (TextUtils.isNullOrEmpty(searchQuery)
                 ? searchDataModel.clear()
```

Comment:
It shouldn't apply, but here we have `Single` that we change to an `Observable` so the function that returns that converted `Single` should have a suffix